packages/iris-grid/src/IrisGridPartitionSelector.tsx

```diff
@@ -294,23 +294,31 @@ class IrisGridPartitionSelector extends Component<
     const { model, partitionConfig } = this.props;
     const { isLoading, partitionFilters, partitionTables } = this.state;
 
-    const { mode, partitions } = partitionConfig;
+    const { partitions } = partitionConfig;
 
+    if (partitionFilters !== null && partitionTables !== null) {
+      partitionFilters.forEach((filter, index) => {
+        partitionTables[index].applyFilter(filter as dh.FilterCondition[]);
+      });
+    }
     const partitionSelectors = model.partitionColumns.map((column, index) => (
       <div key={`selector-${column.name}`} className="column-selector">
         <div className="column-name">{column.name}</div>
-        <TableDropdown
-          className="custom-select-sm"
-          table={partitionTables?.[index]}
-          column={partitionTables?.[index]?.columns[index]}
-          filter={partitionFilters?.[index]}
-          onChange={this.getCachedChangeCallback(index)}
-          selectedValue={mode === 'partition' ? partitions[index] : undefined}
-          disabled={
-            (index > 0 && partitionConfig.mode !== 'partition') || isLoading
-          }
-          formatValue={this.getCachedFormatValueCallback(index)}
-        />
+        {partitionTables?.[index] && (
+          <Picker
+            table={partitionTables[index]}
+            direction="bottom"
+            shouldFlip={false}
+            keyColumn={partitionTables[index].columns[index].name}
+            placeholder={'Loading...' as string}
+            labelColumn={partitionTables[index].columns[index].name}
+            onChange={this.getCachedChangeCallback(index)}
+            defaultSelectedKey={partitions[index] as ItemKey}
+            isDisabled={
+              (index > 0 && partitionConfig.mode !== 'partition') || isLoading
+            }
+          />
+        )}
```

Comment:
As this is used currently, there's a warning logged in console:
```
If you do not provide a visible label, you must specify an aria-label or aria-labelledby attribute for accessibility 
    at $51664a9007a3c6f9$var$Picker (http://localhost:4000/ide/node_modules/.vite/deps/@adobe_react-spectrum.js?v=8e48b768:36010:57)
    at PickerNormalized (http://localhost:4000/ide/@fs/home/bender/dev/deephaven/iris-oss/web-client-ui/main/packages/components/src/spectrum/picker/PickerNormalized.tsx:23:40)
    at Picker (http://localhost:4000/ide/@fs/home/bender/dev/deephaven/iris-oss/web-client-ui/main/packages/jsapi-components/src/spectrum/Picker.tsx:21:26)
    at div
    at div
    at IrisGridPartitionSelector (http://localhost:4000/ide/@fs/home/bender/dev/deephaven/iris-oss/web-client-ui/main/packages/iris-grid/src/IrisGridPartitionSelector.tsx?t=1715716828868:310:9)
```
It's telling us that we're not labelling this component correctly. Instead, the label should be set on the Picker itself, instead of displayed separately in the `column-name` div we have outside of it. Add `label={column.name}` and `labelPosition="side"` to add the label to the `Picker` directly.

With that there will be a slight difference in styles:
![image](https://github.com/deephaven/web-client-ui/assets/4505624/712e532b-05b6-4b90-a10d-b49f093c8370)
![image](https://github.com/deephaven/web-client-ui/assets/4505624/e86caf16-c2f8-4837-8541-df43df64c1af)

Just want to make sure that extra width is okay, @dsmmcken may have a comment about that. 
Test with multiple partitions as well just to make sure it's still groovy:

```
from deephaven import empty_table
t = empty_table(15000).update(["x = i", "y = i", "z=i"])
pt = t.partition_by(["x"])
pt2 = t.partition_by(["x", "y"])
```

Reply:
label difference is fine with me